findbugs/src/java/edu/umd/cs/findbugs/detect/IdentityHashCodeOnValueBasedClass.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * FindBugs - Find Bugs in Java programs
+ * Copyright (C) 2003-2008 University of Maryland
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.OpcodeStack.Item;
+import edu.umd.cs.findbugs.ba.vbc.ValueBasedClassIdentifier;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import edu.umd.cs.findbugs.util.ClassName;
+
+public class IdentityHashCodeOnValueBasedClass extends OpcodeStackDetector {
+
+    private final BugReporter bugReporter;
+
+    public IdentityHashCodeOnValueBasedClass(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
```

Comment:
Done in 8855342.

Apparently I didn't update my branch for almost a year :open_mouth: . Did that now and rebased my changes on top, so the commits listed below are largely the known ones from before - the first new one starts at 4b05469.